test/e2e/packetcapture_test.go

```diff
@@ -922,14 +1055,33 @@ func verifyPacketFile(t *testing.T, pc *crdv1alpha1.PacketCapture, reader io.Rea
 		direction := pc.Spec.Direction
 		switch direction {
 		case crdv1alpha1.CaptureDirectionDestinationToSource:
-			assert.Equal(t, srcIP.String(), ip.DstIP.String())
-			assert.Equal(t, dstIP.String(), ip.SrcIP.String())
+			if srcIP != nil {
+				assert.Equal(t, srcIP.String(), ip.DstIP.String())
+			}
+			if dstIP != nil {
+				assert.Equal(t, dstIP.String(), ip.SrcIP.String())
+			}
 		case crdv1alpha1.CaptureDirectionBoth:
-			assert.Contains(t, []string{srcIP.String(), dstIP.String()}, ip.SrcIP.String())
-			assert.Contains(t, []string{srcIP.String(), dstIP.String()}, ip.DstIP.String())
+			if srcIP != nil && dstIP != nil {
+				assert.Contains(t, []string{srcIP.String(), dstIP.String()}, ip.SrcIP.String())
+				assert.Contains(t, []string{srcIP.String(), dstIP.String()}, ip.DstIP.String())
+			} else {
+				targetIP := srcIP
+				if targetIP == nil {
+					targetIP = dstIP
+				}
+				targetIPStr := targetIP.String()
+				isEgress := ip.SrcIP.String() == targetIPStr
+				isIngress := ip.DstIP.String() == targetIPStr
+				assert.True(t, isEgress || isIngress, "Packet (src=%s, dst=%s) does not involve target Pod %s", ip.SrcIP.String(), ip.DstIP.String(), targetIPStr)
```

Comment:
cam we separate the `srcIP != nil` case from the `dstIP != nil` case? I feel like the code is difficult to understand right now.